pipit/trace.py

```diff
@@ -127,3 +127,130 @@ def comm_matrix(self, output="size"):
             ]
 
         return communication_matrix
+
+    def match_rows(self):
+        if "Matching Index" not in self.events.columns:
+            """
+            Two columns to be added to dataframe:
+            "Matching Index" and "Matching Timestamp"
+
+            Matches dataframe indices and timestamps
+            between corresponding enter and leave rows.
+            """
+            matching_indices = [float("nan")] * len(self.events)
+            matching_times = [float("nan")] * len(self.events)
+
+            enter_leave_df = self.events.loc[
+                self.events["Event Type"].isin(["Enter", "Leave"])
+            ]
+
+            # Filter by Thread/Process
+            for id in set(enter_leave_df["Thread"]):
+                filtered_df = enter_leave_df.loc[enter_leave_df["Thread"] == id]
+
+                stack = []
+                event_types = list(filtered_df["Event Type"])
+                df_indices, timestamps = list(filtered_df.index), list(
+                    filtered_df["Timestamp (ns)"]
+                )
```

Comment:
Why are we creating these lists and not working directly in the dataframe?

Reply:
Working directly in the dataframe is pretty slow, especially when the dataframe gets larger. I can run some tests.

Reply:
Update: For Large Score-P trace available here (https://vampir.eu/downloads/tracefiles), match rows with lists takes about 30 times less running time than with iterrows.